compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5824,10 +5824,78 @@ TR::Register *OMR::Power::TreeEvaluator::retrieveTOCRegister(TR::Node *node, TR:
    return cg->machine()->getRealRegister(TR::RealRegister::gr2);
 }
 
+TR::Register *OMR::Power::TreeEvaluator::sbyteswapEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *tgtRegister = cg->allocateRegister();
+
+   TR::Node *firstNonConversionOpCodeNode = node->getFirstChild();
+   TR::DataType nodeType = firstNonConversionOpCodeNode->getType();
+
+   //Move through descendants until a non conversion opcode is reached,
+   //while making sure all nodes have a ref count of 1 and the types are between 2-8 bytes
+   while (firstNonConversionOpCodeNode->getOpCode().isConversion() &&
```

Comment:
Could you please elaborate on this while loop? Can we prove that it's always correct, eg. conversation and byte swap are commutative?

Reply:
I guess I commented just after the merge. @aviansie-ben could you please still respond?

Reply:
IIUC, this code is attempting to see through patterns of the form:

```
sbyteswap
  i2s
    iload
```

By recognizing this pattern, this can be generated as an `lhbrx` instruction that simply loads the correct part of the value that was actually loaded, with the byteswap as part of the load instead of requiring the byteswap sequence. That being said, I'm not 100% sure on this code myself as I did not write it. As I said in the PR description and commit message, this implementation is coming from the existing code in OpenJ9 that was used to implement the `Short.reverseBytes` recognized method.